src/tracepoint_format_parser.cpp

```diff
@@ -98,12 +105,20 @@ bool TracepointFormatParser::parse(ast::Program *program)
           std::ifstream format_file(format_file_path.c_str());
           if (format_file.fail())
           {
-            std::cerr << "ERROR: tracepoint not found: " << category << ":" << event_name << std::endl;
+            bpftrace.error(std::cerr,
+                           ap->loc,
+                           "tracepoint not found: " + category + ":" +
+                               event_name);
             // helper message:
             if (category == "syscall")
-              std::cerr << "Did you mean syscalls:" << event_name << "?" << std::endl;
+              bpftrace.warning(std::cerr,
+                               ap->loc,
+                               "Did you mean syscalls:" + event_name + "?");
             if (bt_verbose) {
-                std::cerr << strerror(errno) << ": " << format_file_path << std::endl;
+              // Having the location info isn't really useful here, so no
+              // bpftrace.error
+              std::cerr << strerror(errno) << ": " << format_file_path
```

Comment:
Hmm, errno could easily be clobbered here. I read through llvm's libcxx implementation and it seems like errno should still be respected (although I remain skeptical that it's in the spec).

 Can you save a copy of errno right after `if (format_file.fail())` so it's not accidentally clobbered by bpftrace.error() or bpftrace.warning()?

Reply:
Done